cf-agent/Makefile.am

```diff
@@ -130,8 +130,10 @@ if !NT
 noinst_PROGRAMS = manifest_file
 manifest_file_SOURCES = manifest_file.c
 manifest_file_LDADD = ${top_srcdir}/libntech/libutils/libutils.la \
-					  ../libpromises/libpromises.la \
-                      libcf-agent.la
+	../libpromises/libpromises.la \
+	libcf-agent.la \
+	$(SYSTEMD_SOCKET_LIBS)
```

Comment:
The LGTM build got past manifest_file now:

```
[2021-02-23 17:46:23] [build]   CCLD     libcf-agent.la
[2021-02-23 17:46:23] [build]   CCLD     cf-agent
[2021-02-23 17:46:24] [build]   CC       manifest_file.o
[2021-02-23 17:46:25] [build]   CCLD     manifest_file
[2021-02-23 17:46:25] [build] Making all in cf-execd
[2021-02-23 17:46:25] [build]   CC       cf_execd-cf-execd.o
[2021-02-23 17:46:27] [build]   CC       cf-execd.lo
[2021-02-23 17:46:28] [build]   CC       cf-execd-runner.lo
[2021-02-23 17:46:29] [build]   CC       exec-config.lo
[2021-02-23 17:46:30] [build]   CC       execd-config.lo
[2021-02-23 17:46:31] [build]   CC       cf-execd-runagent.lo
[2021-02-23 17:46:32] [build-err] cf-execd-runagent.c: In function ‘HandleRunagentRequest’:
[2021-02-23 17:46:32] [build-err] cf-execd-runagent.c:126:12: warning: variable ‘written’ set but not used [-Wunused-but-set-variable]
[2021-02-23 17:46:32] [build-err]   126 |     size_t written = WriterWrite(json_writer, "{\n\"output\" : \"");
[2021-02-23 17:46:32] [build-err]       |            ^~~~~~~
[2021-02-23 17:46:32] [build-err] cf-execd-runagent.c:57:9: warning: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Wunused-result]
[2021-02-23 17:46:32] [build-err]    57 |         write(conn_fd, error_buf, MIN(sizeof(error_buf), (size_t) length));
[2021-02-23 17:46:32] [build-err]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[2021-02-23 17:46:32] [build]   CCLD     libcf-execd.la
[2021-02-23 17:46:32] [build]   CCLD     cf-execd
[2021-02-23 17:46:33] [build-err] /usr/bin/ld: /opt/src/libpromises/.libs/libpromises.so: undefined reference to `sd_listen_fds'
[2021-02-23 17:46:33] [build-err] collect2: error: ld returned 1 exit status
[2021-02-23 17:46:33] [build-err] make[2]: *** [Makefile:612: cf-execd] Error 1
[2021-02-23 17:46:33] [build-err] make[1]: *** [Makefile:711: all-recursive] Error 1
[2021-02-23 17:46:33] [build-err] make: *** [Makefile:600: all] Error 2
```

It seems you will have to add the appropriate flags to `libpromises/Makefile.am`. I am not sure why the LGTM build is different from others, but we need it to pass.

Reply:
But the above has nothing to do with the changes here, right? Looks like it's something that should be failing on *master* too.

Reply:
See https://github.com/cfengine/libntech/pull/116 and https://github.com/cfengine/core/pull/4519

Reply:
@vpodzime Sorry for the confusion, the warnings there are not what's causing the failure, the relevant parts:

```
[2021-02-23 17:46:23] [build]   CCLD     libcf-agent.la
[2021-02-23 17:46:23] [build]   CCLD     cf-agent
[2021-02-23 17:46:24] [build]   CC       manifest_file.o
[2021-02-23 17:46:25] [build]   CCLD     manifest_file
[2021-02-23 17:46:25] [build] Making all in cf-execd
[2021-02-23 17:46:25] [build]   CC       cf_execd-cf-execd.o
[2021-02-23 17:46:27] [build]   CC       cf-execd.lo
[2021-02-23 17:46:28] [build]   CC       cf-execd-runner.lo
[2021-02-23 17:46:29] [build]   CC       exec-config.lo
[2021-02-23 17:46:30] [build]   CC       execd-config.lo
[2021-02-23 17:46:31] [build]   CC       cf-execd-runagent.lo
[...]
[2021-02-23 17:46:32] [build]   CCLD     libcf-execd.la
[2021-02-23 17:46:32] [build]   CCLD     cf-execd
[2021-02-23 17:46:33] [build-err] /usr/bin/ld: /opt/src/libpromises/.libs/libpromises.so: undefined reference to `sd_listen_fds'
[2021-02-23 17:46:33] [build-err] collect2: error: ld returned 1 exit status
[2021-02-23 17:46:33] [build-err] make[2]: *** [Makefile:612: cf-execd] Error 1
[2021-02-23 17:46:33] [build-err] make[1]: *** [Makefile:711: all-recursive] Error 1
[2021-02-23 17:46:33] [build-err] make: *** [Makefile:600: all] Error 2
```

Again, I think the other Makefiles need the appropriate flags, such as `cf-execd/Makefile.am` and perhaps `libpromises/Makefile.am` - the fix for manifest_file seemed to work.

Reply:
Makes (more) sense. @peckpeck could you please add the missing `$(SYSTEMD_LIBS)`? I think it would be best to add it to *libpromises* which should then make it available for the binaries.

Reply:
done, in fact it works better since it permits builing static and dynamic binaries (probably the cause of the difference between your CI and my build)